File: 2570. Merge Two 2D Arrays by Summing Values/Merge Two 2D Arrays by Summing Values.py
from collections import defaultdict


nums1 = [[1,2],[2,3],[4,5]]
nums2 = [[1,4],[3,2],[4,1]]

temp = defaultdict(int)

for num in nums1 + nums2:
    temp[num[0]] += num[1]

print(sorted(temp.items()))

File: 2570. Merge Two 2D Arrays by Summing Values/MergeTwo2DArraysbySummingValues.java
import java.util.*;

public class MergeTwo2DArraysbySummingValues {
    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
        Map<Integer, Integer> map = new TreeMap<>();

        for (int[] num : nums1) {
            map.put(num[0], map.getOrDefault(num[0], 0) + num[1]);
        }
        for (int[] num : nums2) {
            map.put(num[0], map.getOrDefault(num[0], 0) + num[1]);
        }

        int[][] res = new int[map.size()][2];
        int count = 0;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()){
            res[count][0] = entry.getKey();
            res[count][1] = entry.getValue();
            count++;
        }
        return res;
    }
}
